src/solvers/flattening/boolbv_index.cpp

```diff
@@ -203,8 +202,8 @@ bvt boolbvt::convert_index(const index_exprt &expr)
 
       // add implications
 
-      equal_exprt index_equality;
-      index_equality.lhs()=index; // index operand
+      // index operand
+      equal_exprt index_equality(index, exprt());
```

Comment:
I'm wondering whether it wouldn't be better to construct the equality on conversion?

Reply:
I've now done so, for all the cases that you pointed out.